libknet/tests/api_knet_link_set_enable.c

```diff
@@ -107,7 +107,7 @@ static void test(void)
 
 	printf("Test knet_link_set_enable with incorrect values\n");
 
-	if (knet_link_set_config(knet_h, 1, 0, &src, &dst) < 0) {
+	if (knet_link_set_config(knet_h, 1, 0, KNET_TRANSPORT_UDP, &src, &dst) < 0) {
 		printf("Unable to configure link: %s\n", strerror(errno));
 		knet_host_remove(knet_h, 1);
 		knet_handle_free(knet_h);
```

Comment:
I think it would be very useful to add a few extra links during set_enable with other protocols. This check is good at detecting memory leaks and threading issues without real traffic going when using make check-memcheck and make check-helgrind.